content/en/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: blog
+title: "Kubernetes 1.30: Preventing unauthorised volume mode conversion moves to GA"
+date: 2024-04-30
+slug: prevent-unauthorised-volume-mode-conversion-ga
+---
+
+**Author:** Raunak Pradip Shah (Mirantis)
+
+With the release of Kubernetes 1.30, the feature to prevent the modification of the volume mode
+of a [`PersistentVolumeClaim`](/docs/concepts/storage/persistent-volumes/) that was created from
+an existing VolumeSnapshot in a Kubernetes cluster, has moved to GA!
+
+
+## The problem
+
+The [Volume Mode](/docs/concepts/storage/persistent-volumes/#volume-mode) of a `PersistentVolumeClaim` 
+refers to whether the underlying volume on the storage device is formatted into a filesystem or
+presented as a raw block device to the Pod that uses it.
+
+Users can leverage the `VolumeSnapshot` feature, which has been stable since Kubernetes v1.20,
+to create a `PersistentVolumeClaim` (shortened as `PVC`) from an existing `VolumeSnapshot` in
+the Kubernetes cluster. The `PVC` spec includes a `dataSource` field, which can point to an
+existing `VolumeSnapshot` instance.
+Visit [Create a PersistentVolumeClaim from a Volume Snapshot](/docs/concepts/storage/persistent-volumes/#create-persistent-volume-claim-from-volume-snapshot) 
+for more details on how to create a `PVC` from an existing `VolumeSnapshot` in a Kubernetes cluster.
+
+When leveraging the above capability, there is no logic that validates whether the mode of the
+original volume, whose snapshot was taken, matches the mode of the newly created volume.
+
+This presents a security gap that allows malicious users to potentially exploit an
+as-yet-unknown vulnerability in the host operating system.
+
+There is a valid use case to allow some users to perform such conversions. Typically, storage backup
+vendors convert the volume mode during the course of a backup operation, to retrieve changed blocks 
+for greater efficiency of operations. This prevents Kubernetes from blocking the operation completely
+and presents a challenge in distinguishing trusted users from malicious ones.
+
+## Preventing unauthorised users from converting the volume mode
+
+In this context, an authorised user is one who has access rights to perform `Update`
+or `Patch` operations on `VolumeSnapshotContents`, which is a cluster-level resource.  
+It is upto the cluster administrator to provide these rights only to trusted users
+or applications, like backup vendors.
+Users apart from such unauthorised ones will never be allowed to modify the volume mode
```

Comment:
Do you mean "from such _authorised_ ones" here, referring to the authorized backup vendors use case?

Reply:
Ooh good catch, fixed!

Reply:
Although, US spelling, I think we prefer unauthorized